src/CsvParser.php

```diff
@@ -263,9 +277,11 @@ protected function stripBom(array $row)
     {
         if (strpos($row[0], self::$bom) === 0) {
             $row[0] = str_replace(self::$bom, '', $row[0]);
+            // enclosure must be trimmed manually on first column
+            // when having a BOM
+            $row[0] = trim($row[0], $this->enclosure);
```

Comment:
possible edge case: first column has enclosure, plus the value contains the enclosure character itself, at the start or end of the string - do we care about such "nonesense" :wink: